auth/crypto.go

```diff
@@ -80,18 +77,10 @@ type httpKeySource struct {
 	Mutex      *sync.Mutex
 }
 
-func newHTTPKeySource(ctx context.Context, uri string, opts ...option.ClientOption) (*httpKeySource, error) {
-	var hc *http.Client
-	if ctx != nil && len(opts) > 0 {
-		var err error
-		hc, _, err = transport.NewHTTPClient(ctx, opts...)
-		if err != nil {
-			return nil, err
-		}
-	} else {
+func newHTTPKeySource(ctx context.Context, uri string, hc *http.Client) (*httpKeySource, error) {
+	if hc == nil {
```

Comment:
Once we start initializing hc unconditionally in `auth.go`, remove this fallback.